### PR TITLE
Add chainerx test to dataset_tests

### PR DESCRIPTION
**Pull request by keisukefukuda**
This PR is a part of #8031. It adds ChainerX tests to `test_emtpy_dataset.py` and `test_scatter.py` that use numpy arrays.

### CODE REVIEW

File: tests/chainermn_tests/datasets_tests/test_mnist.py [hunk 1]
```diff
@@ -32,21 +32,22 @@ def __call__(self, x):
...
-        device = testing.get_device(comm.intra_rank, use_chainerx)
-        device.use()
-        model.to_device(device)
-    else:
-        device = testing.get_device(use_chainerx=use_chainerx)
+        # Call CuPy's `Device.use()` to force cudaSetDevice()
+        get_device(comm.intra_rank, False).use()
 
+    device = get_device(comm.intra_rank if use_gpu else None, use_chx)
+    device.use()
```
Comment (belldandyxtq) [inline tests/chainermn_tests/datasets_tests/test_mnist.py]: Unnecessary

File: tests/chainermn_tests/datasets_tests/test_scatter.py [hunk 2]
```diff
@@ -39,13 +40,25 @@ def check_scatter_dataset(self, original_dataset, shuffle=False, root=0):
             # Test the content of scattered datasets
             joined_dataset = sum((sub_dataset[:]
                                   for sub_dataset in sub_datasets), [])
+
+            # NOTE: values in original_dataset must be int-like type
```
Comment (belldandyxtq) [inline tests/chainermn_tests/datasets_tests/test_scatter.py]: I am not sure if I understand, does it mean the values in `original_dataset` and `joined_dataset` must be casted to `int`, because `chainerx` and `numpy` have different behaviors when calling `type()`?

File: tests/chainermn_tests/datasets_tests/test_scatter.py [hunk 3]
```diff
@@ -39,13 +40,25 @@ def check_scatter_dataset(self, original_dataset, shuffle=False, root=0):
...
+            joined_dataset = [int(e) for e in joined_dataset]
+            original_dataset = [int(e) for e in original_dataset]
             self.assertEqual(set(joined_dataset), set(original_dataset))
 
     def test_scatter_dataset(self):
         n = self.communicator.size
 
         for shuffle in [True, False]:
             for root in range(self.communicator.size):
+                self.communicator.mpi_comm.barrier()
```
Comment (belldandyxtq) [inline tests/chainermn_tests/datasets_tests/test_scatter.py]: why do we need a barrier here?

Comment (keisukefukuda) [reply tests/chainermn_tests/datasets_tests/test_scatter.py]: This line is unnecessary. forgot to remove debug code.

Comment (keisukefukuda) [reply tests/chainermn_tests/datasets_tests/test_scatter.py]: fixed.

File: tests/chainermn_tests/datasets_tests/test_scatter.py (hunk 2 above)
Comment (keisukefukuda) [reply tests/chainermn_tests/datasets_tests/test_scatter.py]: Yes, their behaviors are different. Actually, ChainerX and CuPy have the same behavior, which is different from numpy.

```
>>> a = numpy.array([0])
>>> a
array([0])
>>> a[0]
0
>>> type(a[0])
<class 'numpy.int64'>         # <============ (A)

>>> b = chainerx.array([0])
>>> b
array([0], shape=(1,), dtype=int64, device='native:0')
>>> b[0]
array(0, shape=(), dtype=int64, device='native:0')
>>> type(b[0])
<class 'chainerx.ndarray'>   # <============ (B)
```

(A) and (B) show the difference. Because of this, we need to apply `int()` to get the same values between chainerx and numpy.

Comment (belldandyxtq) [reply tests/chainermn_tests/datasets_tests/test_scatter.py]: I see, thank you. Could you rephrase the sentence "values in original_dataset must be int-like type". I think it is a little confusing.
My suggestion is "The values in original_dataset and joined_dataset must be casted to int"

Comment (keisukefukuda) [reply tests/chainermn_tests/datasets_tests/test_scatter.py]: Added comment.

File: tests/chainermn_tests/datasets_tests/test_mnist.py (hunk 1 above)
Comment (keisukefukuda) [reply tests/chainermn_tests/datasets_tests/test_mnist.py]: I consulted ChainerX guys and found that this `device.use()` is necessary. 
ChainerX has its own device context (which means ChainerX calls `cudaSetDevice()` before and after each operation, so the original CUDA device context is preserved. Thus, we need to set both of ChainerX device context and CUDA's global device context to work with MPI.

Comment (belldandyxtq) [reply tests/chainermn_tests/datasets_tests/test_mnist.py]: Hmmm, I see. That is troublesome, and confusing. We have the call `use` on twice on different context to set the same thing.

Comment (belldandyxtq) [reply tests/chainermn_tests/datasets_tests/test_mnist.py]: Same here, the code is unnecessary.

Comment (keisukefukuda) [reply tests/chainermn_tests/datasets_tests/test_mnist.py]: fixed.

**Comment by belldandyxtq** [comment]
pfnci. test this please
